ngi_pipeline/utils/filesystem.py

```diff
@@ -104,27 +104,33 @@ def locate_project(project, subdir="DATA", resolve_symlinks=True,
         return os.path.abspath(project)
     else:
         try:
-            project_data_dir=os.path.join(config["analysis"]["base_root"], config["analysis"]["sthlm_root"], config["analysis"]["top_dir"], subdir)
-            if not os.path.exists(project_data_dir):
-                project_data_dir=os.path.join(config["analysis"]["base_root"], config["analysis"]["upps_root"], config["analysis"]["top_dir"], subdir)
+            base_root = config["analysis"]["base_root"]
+            node_roots = [config["analysis"]["sthlm_root"],
```

Comment:
if I am right this implicitly assumes that `sthlm_root` needs to be in the config while `upps_root` is not mandatory.
Maybe here there should be a get also for `sthlm_root` and raise an exception is both are None